controllers/contact-controller.js

```diff
@@ -308,11 +321,24 @@ const ContactController = Contact => {
     };
 
     const contact = await Contact.create(contactRecord);
+
+    // Link all of the csam reports related to this contact
+    if (body.csamReports && body.csamReports.length) {
+      await Promise.all(
+        body.csamReports.map(r =>
+          CSAMReportController.connectToContact(contact.id, r.id, accountSid),
+        ),
+      );
```

Comment:
Note that the way this and `connectToContact()` are implemented, this will require `2 * length` database calls, where really only one call is required. If our `accountSid` is `123`, the `contactId` is `321`, and we're adding CSAM ids 1 and 2, then we will run these queries:

```
GET * FROM CSAMReports WHERE id = 1 and accountSid = 123
UPDATE CSAMReports WHERE id = 1 SET contactId = 321
GET * FROM CSAMReports WHERE id = 2 and accountSid = 123
UPDATE CSAMReports WHERE id = 2 SET contactId = 321
```

if we implement `connectToContact(contactId, csamReports: array, accountSid)` and then do a bulk update (`UPDATE CSAMReports WHERE id in (1, 2) SET contactId = 321` which I _think_ should be: `CSAMReports.update({contactId = 321}, {where: { id: csamReports }})` then it's just one query.

Reply:
In [this commit](https://github.com/techmatters/hrm/pull/131/commits/5aee95b21dbe2965911dadc28f793ed715fa0999) I'm addressing this issue. Note that here I'm also changing the return of this function not to be the `contact` as it was created, but returning a query to the DB using a method that includes the associations. I don't know if there's other way to do this but let me know if you think this is not correct.